### PR TITLE
Here is My App For Review Consider this the first round.

### PR DESCRIPTION
**Pull request by meezyart**
Here is my rendition of Hacker News.
I'm open to any feedback. Thank you in advance!
@alexwolfe, @pierrebeaucamp, and @alex-kovoy

### CODE REVIEW

File: src/containers/StoryList.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { LINKS_PER_PAGE, MAX_PAGES} from "../constants";
+
+
+export default function StoryList({ history }) {
+  const [storyIds, setStoryIds] = useState([]);
+  let { page } = useParams();
+
+  // for pagination
+  let startStory = LINKS_PER_PAGE * +page,
+    endStory = LINKS_PER_PAGE + LINKS_PER_PAGE * +page - 1;
```
Comment (pierrebeaucamp) [inline src/containers/StoryList.js]: Since this is broken into two lines anyway, I would just use two `let` statements here

File: src/containers/StoryList.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  useEffect(() => {
+    getStoryIds(startStory, endStory).then(
+      data => data && handleStoryIds(data)
+    );
+  }, [page, startStory, endStory]);
+
+  /*
+   * Fixes weird quirk cause by querying the db with orderByKey()
+   * returns a array for for the first call and a object after 2 calls.
```
Comment (pierrebeaucamp) [inline src/containers/StoryList.js]: Some typos in here

File: src/containers/StoryList.js (hunk 1 above)
Comment (pierrebeaucamp) [reply src/containers/StoryList.js]: I'm also not a huge fan of how `endStory` is calculated. I think the following is easier to read, for example:

```js
let endStory = startStory + LINKS_PER_PAGE - 1
```

It's a nice touch to see the unary plus operator here, but I don't think it's needed. `useParams` should make sure that you only get numbers back in the first place.

File: src/containers/StoryList.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  function handleNextPage() {
+    if (+page < 17) {
+      history.push(`/new/${+page + 1}`);
+    } else {
+      page = "0";
+      history.push(`/new/0`);
+    }
+  }
```
Comment (pierrebeaucamp) [inline src/containers/StoryList.js]: Regarding the page 17 thing:

In my opinion, this is a somewhat dangerous way of handling an edge-case like this, for multiple reasons.

1. It's a bad user experience. If we cannot navigate to page 17, why is there a button to do so?
2. Extending on my comment on the routing logic, the lack of a 404 page is creating confusion. If a manually navigate to `/new/17`, I don't expect to end up on page 0. This is easy to miss too. 
3. This is essentially core routing logic which is hidden inside some component that should not know much about routing. This would be very hard to debug for outside collaborators.

My suggestion to fix this would be:

1. Disable the button if the next page is page 17.
2. Add a 404 page in case someone navigates to it directly.

On top of that, is very confusing to me (from reading the code) why page 17 is special. If the API only returns 500 stories, it would be better to encode this somehow in the check (i.e. `if startStory > 500`) or similar. At least add a comment explaining this case, as it's not quite obvious

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+import { render } from "react-dom";
+import App from "./App";
+import * as serviceWorker from "./serviceWorker";
+
+render(<App />,document.getElementById("root"));
+
+// If you want your app to work offline and load faster, you can change
+// unregister() to register() below. Note this comes with some pitfalls.
+// Learn more about service workers: https://bit.ly/CRA-PWA
+serviceWorker.register();
```
Comment (pierrebeaucamp) [inline src/index.js]: What pitfalls? Also the link is outdated.

When something comes with pitfalls, it's always good to see a comment covering those. I.e. something like

> Service Workers can lead to x, which is acceptable since this project does y

File: src/serviceWorker.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    });
+}
+
+export function unregister() {
+  if ('serviceWorker' in navigator) {
+    navigator.serviceWorker.ready.then(registration => {
+      registration.unregister();
+    });
+  }
+}
```
Comment (pierrebeaucamp) [inline src/serviceWorker.js]: A few comments on the file in general:

1. I don't see there being a service worker for this project. Maybe you're planning to add one still, but at this current time the call the `register` just calls `unregister` in turn, etc. I'm somewhat confused by all this as it's stating to be an optional component but it's still being called from index.js. If you don't need it, just remove it.

2. When including code from other sources, it's always good to include a disclaimer at the top of the file

File: src/services/hnAPI.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+export const getStoryIds = async (startCount = '0', endCount = '29') => {
+  const topStoriesRef = firebase
+    .database()
+    .ref(topStoriesUrl)
+    .orderByKey()
+    .startAt(startCount.toString())
+    .endAt(endCount.toString());
+  let results = await topStoriesRef.once("value");
+  return results.val();
```
Comment (pierrebeaucamp) [inline src/services/hnAPI.js]: It would be nice to see some error handling for those methods

File: README.md [hunk 7]
```diff
@@ -1,2 +1,31 @@
...
+Runs the app in the development mode.<br />
+Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+
+The page will reload if you make edits.<br />
+You will also see any lint errors in the console.
+
+### `yarn test`
+
+Launches the test runner in the interactive watch mode.<br />
+See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
```
Comment (pierrebeaucamp) [inline README.md]: I don't know if you're still planning to add tests, but as there are currently none, I would just remove this paragraph

File: src/utils/index.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    return `${interval} hours`;
+  }
+  interval = Math.floor(seconds / 60);
+
+  if (interval > 1) {
+    return `${interval} minutes`;
+  }
+
+  return `${Math.floor(seconds)} seconds`;
+};
```
Comment (pierrebeaucamp) [inline src/utils/index.js]: This is usually the kind of function I don't mind importing from an outside library as there are many intricacies in time that are easy to miss and hard to test for.

About your implementation itself: I would try to avoid so many magic numbers in the arithmetic (or at least provide comments what those numbers are)

File: src/utils/index.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
+
+export const mapComment = commentLen => {
+  if (commentLen > 2) {
+    return `${commentLen} comments`;
+  } else if (commentLen === 1) {
+    return `${commentLen} comment`;
+  } else {
+    return `discuss`;
+  }
```
Comment (pierrebeaucamp) [inline src/utils/index.js]: What happens when there are exactly two comments?

Also, small nitpick: If you return in your `if` branches, you don't need an `else` statement (like you do in `mapToTime` below)

File: src/utils/index.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+};
+
+export function getSourceUrl(url) {
+  const regex = /^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:/\n]+)/;
```
Comment (pierrebeaucamp) [inline src/utils/index.js]: This looks odd, but maybe I'm just reading it wrong. I don't think the whole part with the newlines is needed, that wouldn't happen in a URL.

Also, a comment describing longer regular expressions would be appreciated.

File: src/utils/index.js [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+  let match = regex.exec(url);
+
+  if (!match) return "";
+
+  let sourceUrl = match[1] ? match[1] : "#";
```
Comment (pierrebeaucamp) [inline src/utils/index.js]: Also not quite sure about the two error states of returning `""` and `"#"`. Maybe the second one is on purpose? This is hard to follow, comments would be appreciated.

File: src/utils/index.js [hunk 12]
```diff
@@ -0,0 +1,53 @@
+
+export const mapComment = commentLen => {
```
Comment (pierrebeaucamp) [inline src/utils/index.js]: A few notes about this style of utility functions:

* When functions appear out of context, it is especially important to have comments describing them. Imagine a scenario where you can't see the implementation of a function, only its name and comments. `mapComment` would be pretty confusing in that case.

* Utility functions are also great candidates for unit tests.

File: .gitignore [hunk 13]
```diff
@@ -1,104 +1,23 @@
...
-.rts2_cache_cjs/
-.rts2_cache_es/
-.rts2_cache_umd/
+# production
+/build
 
-# Optional REPL history
-.node_repl_history
+# misc
+.DS_Store
```
[the remark this answers is not in the document]
Comment (meezyart) [reply .gitignore]: Great Feed back< I didn't know about this

File: README.md [hunk 14]
```diff
@@ -1,2 +1,31 @@
...
+
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available Scripts
+
+In the project directory, you can run:
+
+### `yarn start`
+
+Runs the app in the development mode.<br />
```
[the remark this answers is not in the document]
Comment (meezyart) [reply README.md]: Nice removed

File: public/browserconfig.xml [hunk 15]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<browserconfig><msapplication><tile><square70x70logo src="/ms-icon-70x70.png"/><square150x150logo src="/ms-icon-150x150.png"/><square310x310logo src="/ms-icon-310x310.png"/><TileColor>#ffffff</TileColor></tile></msapplication></browserconfig>
```
[the remark this answers is not in the document]
Comment (meezyart) [reply public/browserconfig.xml]: agreed

File: README.md [hunk 16]
```diff
@@ -1,2 +1,31 @@
 # HackerMeez
 A HackerNews React Christmas Clone!
+
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available Scripts
+
+In the project directory, you can run:
```
[the remark this answers is not in the document]
Comment (meezyart) [reply README.md]: Nice Updaed!

File: public/index.html [hunk 17]
```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
```
[the remark this answers is not in the document]
Comment (meezyart) [reply public/index.html]: Nice Catch, I updated the rest of the urls to %PUBLIC_URL%

File: public/browserconfig.xml (hunk 15 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply public/browserconfig.xml]: agreed

File: README.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply README.md]: agreed included

File: public/index.html [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+      href="/android-icon-192x192.png"
+    />
+    <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png" />
+    <link rel="icon" type="image/png" sizes="96x96" href="/favicon-96x96.png" />
+    <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png" />
+    <link rel="manifest" href="/manifest.json" />
+   <link href="https://fonts.googleapis.com/css?family=Roboto:400,700,900&display=swap" rel="stylesheet">
+    <meta name="msapplication-TileColor" content="#ffffff" />
+    <meta name="msapplication-TileImage" content="/ms-icon-144x144.png" />
+    <meta name="theme-color" content="#ffffff" />
```
[the remark this answers is not in the document]
Comment (meezyart) [reply public/index.html]: You are corrrect it was left over... removed

File: public/robots.txt [hunk 19]
```diff
@@ -0,0 +1,2 @@
+# https://www.robotstxt.org/robotstxt.html
+User-agent: *
```
[the remark this answers is not in the document]
Comment (meezyart) [reply public/robots.txt]: good to know!

File: src/components/Comment.js [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+      <CommentHeader {...comment} />
+      <div>
+        <CommentBody {...comment} />
+      </div>
+    </CommentWrapper>
+  );
+};
+
+
+const CommentHeader = ({ kids, by, time, toggleVisible, isVisible }) => {
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: I can dig it!

File: src/components/Comment.js [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+  const [comment, setComment] = useState([]);
+
+  useEffect(() => {
+    getComment(commentId).then(data => data && setComment(data));
+  }, [commentId]);
+
+  return (
+    <CommentWrapper>
+      <CommentHeader {...comment} />
+      <div>
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: Noted.. Will Update. This is great feedback, I love learning new things.

File: src/components/Comment.js [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+  CommentHeaderWrapper,
+  CommentChildren,
+  CommentParent
+} from "../styles/CommentStyle";
+
+export const Comment = ({ commentId }) => {
+  const [comment, setComment] = useState([]);
+
+  useEffect(() => {
+    getComment(commentId).then(data => data && setComment(data));
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: got it!

File: src/components/Header.js [hunk 23]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+import { HeaderWrapper } from '../styles/GlobalStyle'
+
+export const Header = () => {
+  return (
+    <HeaderWrapper>
+      <a href="/">
+        <img src="../apple-icon.png" alt="Hacker-meez" height="20" />
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Header.js]: Sure thing, Nice Catch

File: src/components/Story.js [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+
+  return (
+    <StoryWrapper>
+      <RankWrapper className='rank'>
+        {showRank && <span>{index}.</span>}
+        <VoteButton>&#9650;</VoteButton>
+      </RankWrapper>
+      <div>
+        <div>
+          <LinkWrapper size="1em" color="#262421">
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Story.js]: Got it This makes sense

File: src/serviceWorker.js (hunk 5 above)
Comment (meezyart) [reply src/serviceWorker.js]: Agreed I dont think its needed for this project

File: src/utils/index.js (hunk 8 above)
Comment (meezyart) [reply src/utils/index.js]: Agreed. I was debating in my head bout using these I will just import moment.

File: src/index.js (hunk 4 above)
Comment (meezyart) [reply src/index.js]: I will remove that was a create-react-app boiler plate.

File: src/components/Comment.js (hunk 20 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: 'kds' is actually deconstructed from the API `//{
  "by" : "norvig",
  "id" : 2921983,
  "kids" : [ 2922097, 2922429, 2924562, 2922709, 2922573, 2922140, 2922141 ],
  "parent" : 2921506,
  "text" : "Aw shucks, guys ... you make me blush with your compliments.<p>Tell you what, Ill make a deal: I'll keep writing if you keep reading. K?",
  "time" : 1314211127,
  "type" : "comment"
} */`

File: src/components/Comment.js [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+    </CommentWrapper>
+  );
+};
+
+
+const CommentHeader = ({ kids, by, time, toggleVisible, isVisible }) => {
+  return (
+    <CommentHeaderWrapper>
+      <div>
+        <VoteButton className="vote-btn">&#9650;</VoteButton>
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: added a comment

File: src/components/Comment.js [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+      </LinkWrapper>
+    </CommentHeaderWrapper>
+  );
+};
+
+const CommentBody = ({ text, kids }) => {
+  return (
+    <>
+      <CommentParent>
+        <div dangerouslySetInnerHTML={{ __html: text }} />
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: Added a comment

File: src/components/Comment.js [hunk 27]
```diff
@@ -0,0 +1,69 @@
...
+
+
+const CommentHeader = ({ kids, by, time, toggleVisible, isVisible }) => {
+  return (
+    <CommentHeaderWrapper>
+      <div>
+        <VoteButton className="vote-btn">&#9650;</VoteButton>
+      </div>
+      <LinkWrapper>
+        {by && `by ${by} |`} {time && `${mapToTime(time)} ago`}
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: This was done to prevent the quick flash of "by | NANnum" before the comment flashes in. How do you normally handle this?

File: src/constants/index.js [hunk 28]
```diff
@@ -0,0 +1,2 @@
+export const LINKS_PER_PAGE = 30;
+export const MAX_PAGES = 17;
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/constants/index.js]: Updated

File: src/components/Story.js (hunk 24 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Story.js]: Updated

File: src/containers/Layout.js [hunk 29]
```diff
@@ -0,0 +1,27 @@
...
+    <MainWrapper>
+      <GlobalStyle/>
+      <Header />
+      <section>
+        <BrowserRouter>
+          <Switch>
+            <Route exact path="/" render={() => <Redirect to="/new/0" />} />
+            <Route path="/new/:page" component={StoryList} />
+            <Route path="/item/:itemId" component={StoryDetail} />
+            <Route render={() => <Redirect to="/new/0" />} />
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/Layout.js]: Added a 404 component

File: src/containers/StoryDetail.js [hunk 30]
```diff
@@ -0,0 +1,37 @@
...
+  useEffect(() => {
+    getCommentIds(storyId).then(data => data && setCommentIds(data));
+  }, [storyId]);
+
+  return (
+    <ContentWrapper>
+      <Story storyId={storyId} />
+      <AddCommentBox>
+        <textarea rows="6" cols="60" />
+        <br />
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryDetail.js]: updated

File: src/containers/StoryList.js (hunk 1 above)
Comment (meezyart) [reply src/containers/StoryList.js]: updated nice comments, I'm already learning a lot from you.

File: src/utils/index.js (hunk 9 above)
Comment (meezyart) [reply src/utils/index.js]: updated

File: src/utils/index.js (hunk 8 above)
Comment (meezyart) [reply src/utils/index.js]: Agreed.. but I had issues with achieving the results I wanted with an outside library, so I added comments to accommodate this function.

File: src/services/hnAPI.js (hunk 6 above)
Comment (meezyart) [reply src/services/hnAPI.js]: added

File: src/utils/index.js (hunk 11 above)
Comment (meezyart) [reply src/utils/index.js]: updated

File: src/utils/index.js (hunk 10 above)
Comment (meezyart) [reply src/utils/index.js]: added comments

File: src/components/Comment.js (hunk 20 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: How would you handle this?

File: src/styles/CommentStyle.js [hunk 31]
```diff
@@ -0,0 +1,47 @@
+import styled from "styled-components";
+
+export const AddCommentBox = styled.div`
+  display:flex;
```
Comment (alexwolfe) [inline src/styles/CommentStyle.js]: How might you create this comment box with less styles but with the same result?

File: src/components/Comment.js (hunk 27 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: Nice I will Implament!

File: README.md [hunk 32]
```diff
@@ -1,2 +1,32 @@
...
+
+### `yarn install` or `yarn`
+
+Installs all the pessary dependencies to the project.
```
[the remark this answers is not in the document]
Comment (meezyart) [reply README.md]: when spell check goes wrong lol

File: src/components/Error404.js [hunk 33]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Error404.js]: I will Update

File: src/containers/StoryList.js [hunk 34]
```diff
@@ -0,0 +1,79 @@
...
+   * //TODO MOVE OUT SIDE THIS FUNCTION
+   */
+  if (currentPage >= MAX_PAGES) {
+    history.push(`/404`);
+  }
+  /*
+   * Fixes weird quirk cause by querying the db with orderByKey().
+   * Returns an array for the first call and an object after 2 calls.
+   */
+  function handleStoryIds(ids) {
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryList.js]: Good Idea, I will implament

File: src/containers/StoryList.js [hunk 35]
```diff
@@ -0,0 +1,79 @@
...
+//  api
+import { getStoryIds } from "../services/hnAPI";
+// components
+import { Story } from "../components/Story";
+// styles
+import { StoryListWrapper, ButtonLink } from "../styles/GlobalStyle";
+// constants
+import { LINKS_PER_PAGE, MAX_STORIES, MAX_PAGES } from "../constants";
+
+export default function StoryList({ history }) {
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryList.js]: Depends on the functionality I need,  I always try to use functions. I usually use classes for containers. But If it becomes too large I would try to break things up into smaller functions like in comments.

But I'm open to all techniques, being self-thought it is sometimes hard to pick up best practices unless you are exposed to it.

File: src/components/Story.js [hunk 36]
```diff
@@ -0,0 +1,69 @@
...
+      data => {
+        setStory(data);
+      }).catch(error => {
+        console.log("Story: We are getting this error:");
+        console.error(error);
+      });
+  }, [storyId]);
+  return (
+    <StoryWrapper>
+      <RankWrapper className="rank">
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Story.js]: updated

File: src/components/Story.js [hunk 37]
```diff
@@ -0,0 +1,69 @@
...
+              {")"}
+            </LinkWrapper>
+          )}
+        </div>
+        <LinkWrapper>
+          {story.score && `${story.score} votes`}
+          {"  "}
+          {story.by && <Link to={`/item/${storyId}`}>{`by ${story.by}`}</Link>}
+          {story.by && "  |  "}
+          {story.time && (
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Story.js]: Sounds good I will Break it up

[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Story.js]: way cleaner

File: src/components/Comment.js [hunk 38]
```diff
@@ -0,0 +1,72 @@
...
+      <CommentBody {...comment} />
+    </CommentWrapper>
+  );
+};
+
+const CommentHeader = ({ kids, by, time, toggleVisible, isVisible }) => {
+  return (
+    <CommentHeaderWrapper>
+      {/* &#9650; is Html unicode for the up arrow */}
+      {by && <VoteButton className="vote-btn">&#9650;</VoteButton>}
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: updated

File: src/styles/CommentStyle.js (hunk 31 above)
Comment (meezyart) [reply src/styles/CommentStyle.js]: " The Space betweeeeeeeennn" - dave matthews

File: src/App.js [hunk 39]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import { Layout } from "./containers/Layout.js"
+
+function App() {
+  return <Layout/>;
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/App.js]: Updated

File: src/components/Error404.js (hunk 33 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Error404.js]: updated

File: src/containers/StoryList.js (hunk 34 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryList.js]: Moved!

File: src/services/hnAPI.js [hunk 40]
```diff
@@ -0,0 +1,81 @@
...
+    error => {
+      console.log("getItemById: We are getting this error:");
+      console.error(error);
+      results = error;
+    }
+  );
+  return results;
+};
+
+export const getComment = async commentId => {
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/services/hnAPI.js]: your right I can remove and point to that function instead. In fact, I can get rid of the other two also. I thought it was overkill my self too.

File: src/components/Comment.js (hunk 20 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: adressed with the last check in

File: src/containers/StoryList.js (hunk 3 above)
Comment (meezyart) [reply src/containers/StoryList.js]: implemented

File: src/App.js [hunk 41]
```diff
@@ -0,0 +1,31 @@
...
+      <GlobalStyle />
+      <Header />
+      <section>
+        <BrowserRouter>
+          <Switch>
+            <Route exact path="/" render={() => <Redirect to="/new/0" />} />
+            <Route path="/new/:page" component={StoryList} />
+            <Route path="/item/:itemId" component={StoryDetail} />
+            <Route path="/404" component={Error404} />
+            <Route render={() => <Redirect to="/404" />} />
```
Comment (alex-kovoy) [inline src/App.js]: I do not think that we should be redirecting a user to `/404` page to show that `/404` is not found :) A better UX would be to show `<Error404/>` when nothing else matches the route as shown below.

```jsx
<Switch>
      <Route exact path="/" render={() => <Redirect to="/new/0" />} />
      <Route path="/new/:page" component={StoryList} />
      <Route path="/item/:itemId" component={StoryDetail} />      
      <Route component={Error404} />
</Switch>
```

File: src/selectors/index.js [hunk 42]
```diff
@@ -0,0 +1,9 @@
+export const selectFields = ({ id, by, title, kids, url, time, score }) => ({
```
Comment (alex-kovoy) [inline src/selectors/index.js]: What does this function do and what's the purpose of a dedicated file for that?

File: src/components/Error404.js [hunk 43]
```diff
@@ -0,0 +1,16 @@
...
+import { ContentWrapper,ButtonLink } from '../styles/GlobalStyle'
+
+export const Error404 = ({history}) => {
+  const handleRedirect = () =>{
+    history.push('/new/0')
+  }
+  return (
+    <ContentWrapper>
+      <h1>Page not Found</h1>
+      <ButtonLink onClick={handleRedirect}>
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Error404.js]: sounds good

[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Error404.js]: was able to update them both in global styles

File: src/containers/StoryList.js [hunk 44]
```diff
@@ -0,0 +1,79 @@
...
+  const [storyIds, setStoryIds] = useState([]);
+  const [currentPage, setCurrentPage] = useState({});
+  let { page } = useParams();
+
+  // for pagination
+  let startStory = LINKS_PER_PAGE * +page;
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryList.js]: I love the ideas! I was thinking the same thing. I will look to implement

File: src/components/Comment.js [hunk 45]
```diff
@@ -0,0 +1,83 @@
...
+      <CommentBody {...comment} />
+    </CommentWrapper>
+  );
+};
+
+const CommentHeader = ({ kids, by, time, toggleVisible, isVisible }) => {
+  const author = by ? `by ${by} |` : null;
+  const timeStamp = author && time ? `${mapToTime(time)} ago` : null;
+  /* &#9650; is Html unicode for the up arrow */
+  const voteUpArrow = author ? <VoteButton small>&#9650;</VoteButton> : null;
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: even better!

File: src/containers/Error404.js [hunk 46]
```diff
@@ -0,0 +1,16 @@
...
+import { ContentWrapper,ButtonLink } from '../styles/GlobalStyle'
+
+export const Error404 = ({history}) => {
+  const handleRedirect = () =>{
+    history.push('/new/0')
+  }
+  return (
+    <ContentWrapper>
+      <h1>Page not Found</h1>
+      <ButtonLink onClick={handleRedirect}>
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/Error404.js]: I changed buttonLink to a styled Link in globalStyles.

File: src/containers/StoryList.js [hunk 47]
```diff
@@ -0,0 +1,67 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router";
+//  api
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryList.js]: Agreed!

[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryList.js]: I dont like alot of comments

File: src/services/hnAPI.js [hunk 48]
```diff
@@ -0,0 +1,59 @@
...
+export const version = "/v0/";
+export const topStoriesUrl = `${version}topstories`;
+export const itemUrl = `${version}item/`;
+
+firebase.initializeApp({
+  authDomain: "hacker-news.firebaseio.com",
+  databaseURL: "https://hacker-news.firebaseio.com"
+});
+
+export const getItemById = async itemId => {
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/services/hnAPI.js]: Agreed!! Will clean up.

File: src/components/Comment.js [hunk 49]
```diff
@@ -0,0 +1,83 @@
...
+  // set up to hide the comments
+  const commentCount = kids ? (
+    <>
+      {"  [ "}
+      <button onClick={toggleVisible}>
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: I started down this path but decided against, sticking to scope. I left it in because It looked cool as a display marker. I will remove the functionality.

File: src/App.js (hunk 41 above)
Comment (meezyart) [reply src/App.js]: Nice Catch

File: src/selectors/index.js (hunk 42 above)
Comment (meezyart) [reply src/selectors/index.js]: I had removed the usage for this I will clean this up.

File: src/containers/StoryList.js [hunk 50]
```diff
@@ -0,0 +1,79 @@
...
+      {storyIds.map((storyId, index) => (
+          <Story
+            key={storyId}
+            storyId={storyId}
+            showRank={true}
+            index={startStory + index + 1}
+          />
+        ))}
+        {/* Hides button when the 500th article is reached */}
+      <ButtonLink hidden={endStory > MAX_STORIES} onClick={handleNextPage}>
```
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryList.js]: Nice! Waay cleaner.

File: src/components/Comment.js (hunk 45 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: Learning so much!

[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: Actually a string didn't work, but this did `const UP_ARROW = '▲';`

File: src/components/Error404.js (hunk 43 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Error404.js]: Updated

File: src/services/hnAPI.js (hunk 48 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/services/hnAPI.js]: Reformatted the Api, simplified it more.

File: src/components/Comment.js (hunk 49 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/components/Comment.js]: Ended Up adding in this functionality.

File: src/App.js (hunk 41 above)
Comment (meezyart) [reply src/App.js]: Updated and added some regex to it too.

File: src/containers/StoryList.js (hunk 44 above)
[the remark this answers is not in the document]
Comment (meezyart) [reply src/containers/StoryList.js]: I implemented a fix

**Comment by meezyart** [comment]
Can I request another review, for learning purpose only...if you have time? At your earliest convenience.
